### ISSUE TITLE
Change in "status" label

### ISSUE THREAD

**Issue by AndresSepulveda** [OpenDrift/opendrift#615]
Hi,

 I just noticed that the valid values for the "status" variable change depending on the available options.

If no particles fall under the category "stranded" of "released on land", the available options are

    0           1             2
active missing_data retired

however, if particles can be stranded or are being released on land, the values for the "status" variable are:

 0                 1                     2                    3

active seeded_on_land missing_data retired

Wouldn't it be better to assign a fixed value for each category?

Regards,

Andres

**Comment by knutfrode** [comment]
Hi,

With fixed value for each category, then each module would need to maintain a separate definition/mapping of status number and description, and it would be difficult to maintain backwards compatibility if these change.

With dynamic status categories this is much simpler, and a model developer can simply throw in a line with
```
self.deactivate_elements(condition, reason='description')
```
without needing to update anything else.

After the simulation, the list `o.status_categories` contains the names in the order they appeared, and you may e.g. get the actual number for `stranded` with:
```
>>> strandnum = o.status_categories.index('stranded')
```

This is stored in the netCDF file is according to the CF-convention for `Flags`:
https://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#flags

**Comment by AndresSepulveda** [comment]
Ok, I understand. 
Is the order of the categories fixed? That is, if there are three values, they will always be

active  missing_data retired

?

With four values it can be more complicated, Seems the categories could be

active seeded_on_land missing_data retired

or

active stranded missing_data retired

is this correct?

**Comment by knutfrode** [comment]
No, the order is not fixed. Only categories which have been used are listed, and in the order in which they first appear during the actual simulation.
Only status=0 (active) is fixed.
